### PR TITLE
Make Determinant commutative

### PR DESCRIPTION
**Pull request by eric-wieser**
This brings it in line with sympy.matrices.expressions.Trace, which is also marked commutative.

#### References to other Issues or PRs

Related to #19308. Note that after this change, `Determinant` will inherit the same bug that `Trace` has, namely #19353. Dealing with that seems out of scope for this PR.

#### Brief description of what is fixed or changed
```python
>>> Determinant.is_commutative
True  # previously None
```

#### Release Notes

* matrices
  * `Determinant` is now considered commutative.

### CODE REVIEW

**Comment by sylee957** [comment]
Can you add some tests in test_determinant.py?

**Comment by eric-wieser** [comment]
Added a test

**Comment by sylee957** [comment]
I think that it’s better to test whether is_commutative returns True for a certain matrix

**Comment by eric-wieser** [comment]
Is that test even interesting? Since `is_commutative` is a static attribute, `Determinant(m).is_commutative` is the same as `Determinant.is_commutative`.

Adding a test like `assert Determinant.is_commutative is True` seems pointless to me.

**Comment by oscarbenjamin** [comment]
For now it's a static attribute but that could be changed. The important point is that the `is_commutative` property is needed on the instance somehow.

**Comment by eric-wieser** [comment]
Presumably though the fact it's called `is_commutative` could also be changed, while we're listing possible changes. Isn't the best thing therefore to test that it _behaves_ commutatively, which is exactly what the test I added does?

**Comment by oscarbenjamin** [comment]
I would test both :)

File: sympy/matrices/expressions/tests/test_determinant.py [hunk 1]
```diff
@@ -38,3 +38,9 @@ def test_eval_determinant():
 def test_refine():
     assert refine(det(A), Q.orthogonal(A)) == 1
     assert refine(det(A), Q.singular(A)) == 0
+
+
+def test_commutative():
+    det_a = Determinant(A)
+    det_b = Determinant(B)
+    assert det_a * det_b == det_b * det_a
```
Comment (eric-wieser) [inline sympy/matrices/expressions/tests/test_determinant.py:46]:
```suggestion
    assert det_a.is_commutative
    assert det_b.is_commutative
    assert det_a * det_b == det_b * det_a
```

**Comment by eric-wieser** [comment]
Added :)

**Pull request merged**